pkg/compute/kube/client.go

```diff
@@ -673,6 +673,16 @@ func (c *client) serviceToMeshServices(svc corev1.Service) []service.MeshService
 		// use port.appProtocol if specified, else use port protocol
 		meshSvc.Protocol = pointer.StringDeref(portSpec.AppProtocol, protocol)
 
+		// If app protocol was not detected and the protocol is TCP, use HTTP
+		// as the default app protocol.
+		// This is required for backward compatibility where users may be relying
+		// on implicit HTTP protocol selection.
+		if meshSvc.Protocol == "" && portSpec.Protocol == corev1.ProtocolTCP {
```

Comment:
I think this conditional should probably be relaxed. As it stands, we still have error/warning log containing the original protocol in the `protocol: udp` case.

Reply:
Sorry, I didn't follow your comment. Could you clarify what scenario results in which error?

Reply:
Your assessment is correct. An error in LDS will indicate unsupported/unknown app protocols. I'll add a comment regarding the empty protocol case.